yt/frontends/ramses/fields.py

```diff
@@ -418,8 +418,16 @@ def create_cooling_fields(self) -> bool:
         def _create_field(name, interp_object, unit):
             def _func(field, data):
                 shape = data["gas", "temperature_over_mu"].shape
+                # Ramses assumes a fraction X of Hydrogen within the non-metal gas.
+                # It has to be corrected by metallicity.
+                Z = data["gas", "metallicity"]
+                nH = (1 - _Y) * (1 - Z) * data["gas", "density"] / mh
```

Comment:
`nh` here will have dimensions of of 1/length**3 -- any need to force a particular unit before using the numpy functions below?